### PR TITLE
✨ (data table filter) hide "Show selection only" toggle if appropriate

### LINKED ISSUE #2790: Data table: Default to showing the selection only if entity selection is disabled
## Description

If entity selection is disabled, the chart might display a subset of entities, whereas the data table displays all entities by default.

## Expected behaviour

Default to showing the selection only if entity selection is disabled.

## Steps to reproduce

1. Go to https://ourworldindata.org/grapher/out-of-school-girls-of-primary-school-age-by-world-region
2. Note that the charts shows regions only and entity selection is disabled
3. Click on 'Table'
4. Note that the table shows all available data, including countries not displayed in the chart

## Screenshots

Screenshot 2023-10-18 at 16 03 09
Screenshot 2023-10-18 at 16 03 15

### LINKED ISSUE #2821: Data table: Hide "Show selection only" toggle when entity selection is disabled
## Description

There is no need to show the "Show selection only" toggle when entity selection is disabled.

## Steps to reproduce

Examples:

- https://ourworldindata.org/grapher/correlation-income-and-life-satisfaction?tab=table
- https://ourworldindata.org/grapher/corporate-investment-in-artificial-intelligence-by-type?tab=table
- https://ourworldindata.org/grapher/average-oecd-education-expenditure-by-source-of-funding-gdp?tab=table
- https://ourworldindata.org/grapher/human-heights-over-the-long-run?tab=table
- https://ourworldindata.org/grapher/the-price-for-lighting-per-million-lumen-hours-in-the-uk-in-british-pound?tab=table

### PR DESCRIPTION
**Pull request by sophiamersmann**
### Summary

- The "Show selection only" toggle only makes sense in concert with the entity selector
- If entity selection is disabled, the "Show selection only" toggle is hidden
- If entity selection is disabled, the data table shows the current selection only
     - Unless Grapher has a map tab, in which case the table shows all entities
     - Unless Grapher has no chart tab
     - Unless the selection is empty

### Example charts

- http://staging-site-improve-show-selection-only-toggle/grapher/out-of-school-girls-of-primary-school-age-by-world-region?tab=table
    - Grapher doesn't have an entity selector -> "Show selection only" toggle is hidden and the current selection is applied
- http://staging-site-improve-show-selection-only-toggle/grapher/population-census-world-bank?tab=table
    - Grapher doesn't have a chart tab, thus doesn't have an entity selector -> "Show selection only" toggle is hidden (but the current selection is not applied since we want to show all entities for maps)
- http://staging-site-improve-show-selection-only-toggle/grapher/sources-population-dataset
    - Grapher only has a table tab -> "Show selection only" toggle is hidden (but the table is not filtered by default)

### References

- fixes #2821 
- fixes #2790 

## Summary by CodeRabbit

- **New Features**
  - Enhanced SVG comparison tool with HTML report generation for visual differences.
  - Improved accessibility across the site by adding `aria-label` attributes to interactive elements.
  - Added "View live" link on document preview pages based on publication status.
  - Implemented new query parameter combinations for chart configurations in testing tools.
  - Introduced a new `cartesian` utility function for computing the cartesian product of arrays.

- **Bug Fixes**
  - Fixed an issue with Google Docs parsing by adjusting character replacements in API router.
  - Corrected image URL handling by using `encodeURIComponent` for better consistency.

- **Documentation**
  - Updated documentation to reflect changes in SVG comparison and testing tools.

- **Refactor**
  - Refactored several components in the admin site client for better maintainability.
  - Improved the modularity and readability of the `export-graphs.js` script.
  - Refined computed properties in `Grapher.tsx` for clearer entity selection logic.

- **Style**
  - Adjusted icon margin in `GdocsEditLink.tsx` for better visual layout.

- **Chores**
  - Updated the environment variable format in SVG comparison workflow.
  - Cleaned untracked files in SVG repository during testing.

- **Revert**
  - Removed "View changed SVGs" link from the admin site test index page.

### CODE REVIEW

File: packages/@ourworldindata/grapher/src/controls/SettingsMenu.tsx [hunk 1]
```diff
@@ -272,6 +268,10 @@ export class SettingsMenu extends React.Component<{
         return type.replace(/([A-Z])/g, " $1")
     }
 
+    @computed get selectionArray(): SelectionArray {
+        return makeSelectionArray(this.manager)
+    }
```
Comment (danyx23) [inline packages/@ourworldindata/grapher/src/controls/SettingsMenu.tsx:273]: This introduces a bug AFAICS - the mobx dependency is only on this.manager and so when the shared selection is changed outside of this selectionArray will not be recomputed

**Review by danyx23 — CHANGES_REQUESTED** [review]
This introduces a bug where the "Show selection only" is hidden in cases where [it should be shown[(http://staging-site-improve-show-selection-only-toggle/grapher/life-expectancy). I think the cause is the way selectionArray is implemented on SettingsMenu but I haven't investigated in detail :)

**Review by danyx23 — CHANGES_REQUESTED** [review]
This introduces a bug where the "Show selection only" is hidden in cases where [it should be shown[(http://staging-site-improve-show-selection-only-toggle/grapher/life-expectancy). I think the cause is the way selectionArray is implemented on SettingsMenu but I haven't investigated in detail :)

**Comment by sophiamersmann** [comment]
Ohh, good catch! It turned out to be exactly what you said, but `showEntitySelectionToggle` (that is defined on Grapher and returned `EntitySelectionToggle.shouldShow(this)`) was the bad guy. Updating this method fixes the bug.

I think I want to look at the `selectionArray` in a separate PR. Most chart types use the same exact function call, so I would like to update the code in all instances, but don't want to fold this change into this PR in case there are any issues.

**Comment by sophiamersmann** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #3060** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3060?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #3071** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3071?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3060?utm_source=stack-comment).

**Comment by danyx23** [comment]
Hey, sorry for the long radio silence, I must have missed the notification that you wanted review again. 

I found another issue - on Explorers the option to filter to the selection makes sense but it is no longer avialable (e.g. https://ourworldindata.org/explorers/population-and-demography?tab=table&facet=none&showSelectionOnlyInTable=1&Metric=Population&Sex=Both+sexes&Age+group=Total&Projection+Scenario=None&country=CHN~IND~USA~IDN~PAK~NGA~BRA~JPN ). I think it would be good to have it there again.

**Comment by sophiamersmann** [comment]
### Merge activity

* **Jan 26, 3:33 AM**: @sophiamersmann started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3060).

**Pull request merged**